python/llm/src/bigdl/llm/transformers/models/llama.py

```diff
@@ -59,21 +59,23 @@ def repeat_kv(hidden_states: torch.Tensor, n_rep: int) -> torch.Tensor:
 KV_CACHE_ALLOC_BLOCK_LENGTH = 256
 
 
-def get_ipex_version():
+_ipex_version = None
 
-    if importlib.util.find_spec("intel_extension_for_pytorch") is not None:
-        import intel_extension_for_pytorch as ipex
-        return ipex.__version__
-    else:
-        return None
 
+def get_ipex_version():
+
+    global _ipex_version
+    if _ipex_version is not None:
+        return _ipex_version
 
-ipex_version = get_ipex_version()
+    import intel_extension_for_pytorch as ipex
+    _ipex_version = ipex.__version__
+    return _ipex_version
 
 
 def llama_rms_norm_forward(self, hidden_states):
     if hidden_states.device.type == "xpu" and not (self.training and hidden_states.requires_grad):
-        if ipex_version == "2.0.110+xpu":
+        if get_ipex_version() == "2.0.110+xpu":
```

Comment:
do we need to fix it for other models? @qiuxin2012

Reply:
We only have this check in llama as an easy way to test internal IPEX on AutoTP. We might need this in other models when the new IPEX come out. (Or we can ask them to keep it backward compatible.)